src/core/NgenSimulation.cpp

```diff
@@ -0,0 +1,263 @@
+#include <NgenSimulation.hpp>
+#include <NGenConfig.h>
+#include <Logger.hpp>
+
+#if NGEN_WITH_ROUTING
+#include "bmi/Bmi_Py_Adapter.hpp"
+#endif // NGEN_WITH_ROUTING
+
+#include "parallel_utils.h"
+
+NgenSimulation::NgenSimulation(
+    std::shared_ptr<realization::Formulation_Manager> formulation_manager,
+    std::vector<std::shared_ptr<ngen::Layer>> layers,
+    std::unordered_map<std::string, int> catchment_indexes,
+    std::unordered_map<std::string, int> nexus_indexes,
+    int mpi_rank,
+    int mpi_num_procs
+                               )
+    : catchment_formulation_manager_(formulation_manager)
```

Comment:
Is there a reason we need `NgenSimulation` to have a reference to the formulation manger? If the governing idea of creating the class is to make it portable and reusable, tying it to a `Formulation_Manager` this way seems like it has too much responsibility. Most of the need in the current code references just the `Simulation_Time_Object`, so the formulation manager could choose whether to share its simulation time with the `NgenSimulation` or create a copy. The only other places it's needed by `NgenSimulation` is in the routing, but the check for whether to run is already done in `NGen.cpp`, and the T-Route config file could just be another parameter.